### DISCUSSION TITLE
Documenting native arrays (et al.)

### DISCUSSION THREAD

**Discussion by 2colours** [perl6/doc#4179] [Ideas]
Hi,

this is something I'm not sure how to go about... the situation is the following. Say you have
```raku
my uint @native-array;
```
This is something you can deduce from the docs you can do. The type of this is, however
```raku
@native-array.WHAT # (array[uint])
```
Mind the lowercase `array`. This is not the [plain old Array](https://docs.raku.org/type/Array) we know from the docs - this is a native array, [as defined here](https://github.com/rakudo/rakudo/blob/main/src/core.c/native_array.pm6).

If you know something like this exists, you can find [pointers to it](https://docs.raku.org/language/nativetypes#index-entry-native_array) but it's far from being documented as a type.

At the same time, I'm also not sure how much this is considered "standard Raku" and how much "just how Rakudo works".

How do you think we should relate to this?

**Comment by coke** [comment]
Are native arrays in roast?

**Reply by Kaiepi** [reply]
Yes, under [`S09-typed-arrays`](https://github.com/Raku/roast/tree/master/S09-typed-arrays) primarily.
